project-koku/nise
# [COST-4407] Add service account support

* Add service account support as basic auth is becoming EOL

### Test
In order to test checkout the branch.
- Build the package `make install`
- Create a service account
- Add environment variables `HCC_SERVICE_ACCOUNT_ID`, `HCC_SERVICE_ACCOUNT_SECRET`, and `HCC_TOKEN_URL` (depending on your environment
- Set your environment
- Run a command to send OCP data to the service: `nise report ocp --insights-upload https://console.redhat.com/api/ingress/v1/upload -s 2023-11-01 --ocp-cluster-id my-test-upload`
- Check listener logs:
```
[2023-11-03 15:50:09,594] INFO None 1 {'message': "Extracting Payload for msg: {'account': '', 'category': 'tar', 'content_type': 'application/vnd.redhat.hccm.tar+tgz', 'metadata': {'reporter': '', 'stale_timestamp': '0001-01-01T00:00:00Z'}, 'request_id': '09740dcb4f05418c84dd0cc6ccabebc6', 'principal': 'wwwww', 'org_id': 'yyyyyy', 'service': 'hccm', 'size': 974, 'url': 'https'
[2023-11-03 15:50:09,685] INFO None 1 {'message': 'Payload with the request id 09740dcb4f05418c84dd0cc6ccabebc6 from cluster my-test-upload is part of the report with manifest id 4e0e0dde-228e-4166-960f-de536237883e', 'tracing_id': '09740dcb4f05418c84dd0cc6ccabebc6', 'account': '', 'org_id': 'yyyyyyy', 'request_id': '09740dcb4f05418c84dd0cc6ccabebc6', 'cluster_id': 'my-test-upload', 'manifest_uuid': '4e0e0dde-228e-4166-960f-de536237883e'}
[2023-11-03 15:50:09,703] WARNING None 1 {'message': 'Recieved unexpected OCP report from my-test-upload', 'tracing_id': '4e0e0dde-228e-4166-960f-de536237883e', 'account': '', 'org_id': 'yyyyyy', 'request_id': '09740dcb4f05418c84dd0cc6ccabebc6', 'cluster_id': 'my-test-upload', 'manifest_uuid': '4e0e0dde-228e-4166-960f-de536237883e'}
```
